src/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.SecKeyRef.Export.cs

```diff
@@ -0,0 +1,202 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.Apple;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static partial class AppleCrypto
+    {
+        private const string OidPbes2 = "1.2.840.113549.1.5.13";
+        private const string OidPbkdf2 = "1.2.840.113549.1.5.12";
+        private const string OidSha1 = "1.3.14.3.2.26";
+        private const string OidTripleDesCbc = "1.2.840.113549.3.7";
+
+        [DllImport(Libraries.AppleCryptoNative)]
+        private static extern int AppleCryptoNative_SecKeyExport(
+            SafeSecKeyRefHandle key,
+            int exportPrivate,
+            SafeCreateHandle cfExportPassword,
+            out SafeCFDataHandle cfDataOut,
+            out int pOSStatus);
+
+        internal static DerSequenceReader SecKeyExport(
+            SafeSecKeyRefHandle key,
+            bool exportPrivate)
+        {
+            // Apple requires all private keys to be exported encrypted, but since we're trying to export
+            // as parsed structures we will need to decrypt it for the user.
+            const string ExportPassword = "DotnetExportPassphrase";
+
+            SafeCreateHandle exportPassword = exportPrivate
+                ? CoreFoundation.CFStringCreateWithCString(ExportPassword)
+                : new SafeCreateHandle();
```

Comment:
Can we avoid allocating unnecessary objects like this with some restructuring?  Or maybe you could use a singleton invalid instance?